dartagnan/src/test/java/com/dat3m/dartagnan/AbstractDartagnanTest.java

```diff
@@ -64,16 +66,79 @@
         this.wmm = wmm;
         this.settings = settings;
     }
-    
-    @Test()
+
+    //@Test
+    public void testCombined() {
+        // Compares Refinement and Assume solver on litmus tests
+        // Replaces location-based assertions by ...
+        final boolean REPLACE_BY_TRUE = true;
+        // ... to allow Refinement to work
+
+        Result res = Result.UNKNOWN;
+        try (SolverContext ctx = TestHelper.createContext();
+             ProverEnvironment prover1 = ctx.newProverEnvironment(ProverOptions.GENERATE_MODELS);
+             ProverEnvironment prover2 = ctx.newProverEnvironment(ProverOptions.GENERATE_MODELS))
+        {
+            Program program = new ProgramParser().parse(new File(path));
+            if (program.getAss() != null) {
+                program.setAss(program.getAss().removeLocAssertions(REPLACE_BY_TRUE));
+                VerificationTask task = new VerificationTask(program, wmm, target, settings);
+                res = runAnalysisTwoSolvers(ctx, prover1, prover2, task);
+            }
+        } catch (Exception e) {
+            fail(e.getMessage());
+        }
+
+        try (SolverContext ctx = TestHelper.createContext();
+             ProverEnvironment prover = ctx.newProverEnvironment(ProverOptions.GENERATE_MODELS))
+        {
+            Program program = new ProgramParser().parse(new File(path));
+            if (program.getAss() != null) {
+                program.setAss(program.getAss().removeLocAssertions(REPLACE_BY_TRUE));
+                VerificationTask task = new VerificationTask(program, wmm, target, settings);
+                assertEquals(res, Refinement.runAnalysisGraphRefinement(ctx, prover,
+                        RefinementTask.fromVerificationTaskWithDefaultBaselineWMM(task)));
+            }
+        } catch (Exception e) {
+            fail(e.getMessage());
+        }
+    }
+
+    @Test
     public void test() {
         try (SolverContext ctx = TestHelper.createContext();
              ProverEnvironment prover1 = ctx.newProverEnvironment(ProverOptions.GENERATE_MODELS);
              ProverEnvironment prover2 = ctx.newProverEnvironment(ProverOptions.GENERATE_MODELS))
         {
             Program program = new ProgramParser().parse(new File(path));
-            VerificationTask task = new VerificationTask(program, wmm, target, settings);
-            assertEquals(expected, runAnalysisTwoSolvers(ctx, prover1, prover2, task));
+            if (program.getAss() != null) {
+                // program.setAss(program.getAss().removeLocAssertions(true)); // TEST CODE for comparing!
```

Comment:
why do we need to compare anything here is we have an explicit method to compare both analysis?

Reply:
I removed the commented code